File: 3course OOP/CPP Labs/laba1/Ceh.cpp
#include "Ceh.h"

#include <iostream>

using namespace std;

Ceh::Ceh():name(NULL),director(NULL),workers(0) {
	cout << "simple create\n";
}
Ceh::Ceh(char* name, char* director, int workers) : name(name), director(director), workers(workers) {
	cout << "Constructor with params\n";
	/*this->name = name;
	this->director = director;
	this->workers = workers;*/
}
Ceh::Ceh(const Ceh& ceh) {
	cout << "Construct-copy: " << ceh.name << endl;
	this->name = ceh.name;
	this->director = ceh.director;
	this->workers = ceh.workers;
}
Ceh::~Ceh() {
	cout << "DELETED:  \n";
	this->showAllParams();
}

char* Ceh::getName() {
	return this->name;
}
char* Ceh::getDirector() {
	return this->director;
}
int Ceh::getWorkersCount() {
	return this->workers;
}

void Ceh::setName(char* name) {
	cout << "setName:   " << name << endl;
	this->name = name;
}
void Ceh::setDirector(char* director) {
	cout << "setDirector:   " << director << endl;
	this->director = director;
}
void Ceh::setWorkerCount(int count) {
	cout << "setWorkerCount:   " << count << endl;
	this->workers = count;
}

void Ceh::showAllParams() {
	cout << "Ceh values: "
		<< "[Name = " << this->name
		<< "], [Director = " << this->director
		<< "], [Workers = " << this->workers << "]\n";
}

File: 3course OOP/CS Labs/laba1/zavd1/Ceh.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace laba1
{
    class Ceh
    {
        private string name;
        private string director;
        private int workers;

        public Ceh()
        {
            Console.WriteLine("Simple create");
            name = "";
            director = "";
            workers = 0;
        }
        public Ceh(string name, string director, int workers)
        {
            Console.WriteLine("Construct with params");
            this.name = name;
            this.director = director;
            this.workers = workers;
        }
        public Ceh(Ceh ceh)
        {
            Console.WriteLine("Construct-copy");
            this.name = ceh.name;
            this.director = ceh.director;
            this.workers = ceh.workers;
        }

        public string getName()
        {
            return this.name;
        }
        public string getDirector()
        {
            return this.director;
        }
        public int getWorkersCount()
        {
            return this.workers;
        }

        public void setName(string name)
        {
            this.name = name;
        }
        public void setDirector(string director)
        {
            this.director = director;
        }
        public void setWorkerCount(int count)
        {
            this.workers = count;
        }

        public void showAllParams(){
            Console.WriteLine(
                String.Format("Name={0}, Director={1}, Workers={2}",
                this.name, this.director, this.workers));
        }
    }
}
